plugins/destination/managed_writer.go

```diff
@@ -150,17 +145,9 @@ func (p *Plugin) writeManagedTableBatch(ctx context.Context, sourceSpec specs.So
 	}
 	p.workersLock.Unlock()
 
-	sourceColumn := &schema.Text{}
-	_ = sourceColumn.Set(sourceSpec.Name)
-	syncTimeColumn := &schema.Timestamptz{}
-	_ = syncTimeColumn.Set(syncTime)
 	for r := range res {
-		// this is a check to keep backward compatible for sources that are not adding
-		// source and sync time
-		if len(r.Data) < len(tables.Get(r.TableName).Columns) {
-			r.Data = append([]schema.CQType{sourceColumn, syncTimeColumn}, r.Data...)
-		}
-		workers[r.TableName].ch <- r.Data
+		tableName := r.Schema().Metadata().Values()[0]
```

Comment:
```suggestion
		md := r.Schema().Metadata()
		nameIndex := md.FindKey("table")
		tableName := md.Values()[nameIndex]
```

Let's use `FindKey` here so that we can add more metadata later without the possibility of it causing a bug. We may also want to suggest adding some more methods to the `Metadata` struct so we don't have to do this everywhere

Reply:
Oh I see we have a `TableNameFromSchema` function, maybe we can use that here?

Reply:
Added `GetValue` to `MetaData` via https://github.com/cloudquery/arrow/pull/4